### ISSUE TITLE
Improve gateways in process

### ISSUE THREAD

**Issue by tjololo** [Altinn/app-lib-dotnet#192]
### Description

How gateways decides what flows to follow is not solved in a consistent way. 
We should have a new look at gateways in our process engine.

### Additional Information
Related epic: https://github.com/Altinn/app-template-dotnet/issues/170

#### Improved gateways given new proposal for process definition
If we decide to implement the new process definition pattern we could make it possible for developers to define expressions on flows exiting gateways to make decisions as to what flow to follow.

This also requires frontend to pass the action chosen by the user to the backend on processNext

Example process definition

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL"
  xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI"
  xmlns:dc="http://www.omg.org/spec/DD/20100524/DC"
  xmlns:altinn="http://altinncdn.no/schemas/1.0/bpmn"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns:di="http://www.omg.org/spec/DD/20100524/DI"
  xmlns:modeler="http://camunda.org/schema/modeler/1.0" id="Definitions_1exv6jf"
  targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="5.5.0"
  modeler:executionPlatform="Camunda Platform" modeler:executionPlatformVersion="7.18.0">
  <bpmn:process id="Process_1om6n79" isExecutable="true">
    <bpmn:startEvent id="Start">
      <bpmn:outgoing>SequenceFlow_1</bpmn:outgoing>
    </bpmn:startEvent>
    <bpmn:task id="Task1" name="Task1:Data">
      <bpmn:extensionElements>
        <altinn:properties>
          <altinn:actions>
            <altinn:action id="submit" name="Send inn" />
          </altinn:actions>
          <altinn:tasktype id="data" />
        </altinn:properties>
      </bpmn:extensionElements>
      <bpmn:incoming>SequenceFlow_1</bpmn:incoming>
      <bpmn:incoming>SequenceFlow_5</bpmn:incoming>
      <bpmn:outgoing>SequenceFlow_2</bpmn:outgoing>
    </bpmn:task>
    <bpmn:task id="Task2" name="Task2:Confirm">
      <bpmn:extensionElements>
        <altinn:properties>
          <altinn:actions>
            <altinn:action id="confirm" name="Bekreft" />
            <altinn:action id="reject" name="Avslå" />
          </altinn:actions>
          <altinn:tasktype id="confirm" />
        </altinn:properties>
      </bpmn:extensionElements>
      <bpmn:incoming>SequenceFlow_2</bpmn:incoming>
      <bpmn:outgoing>SequenceFlow_3</bpmn:outgoing>
    </bpmn:task>
    <bpmn:exclusiveGateway id="Gateway_1">
      <bpmn:incoming>SequenceFlow_3</bpmn:incoming>
      <bpmn:outgoing>SequenceFlow_4</bpmn:outgoing>
      <bpmn:outgoing>SequenceFlow_5</bpmn:outgoing>
    </bpmn:exclusiveGateway>
    <bpmn:endEvent id="End">
      <bpmn:incoming>SequenceFlow_4</bpmn:incoming>
    </bpmn:endEvent>
    <bpmn:sequenceFlow id="SequenceFlow_1" sourceRef="Start" targetRef="Task1" />
    <bpmn:sequenceFlow id="SequenceFlow_2" sourceRef="Task1" targetRef="Task2" />
    <bpmn:sequenceFlow id="SequenceFlow_3" sourceRef="Task2" targetRef="Gateway_1" />
    <bpmn:sequenceFlow id="SequenceFlow_4" sourceRef="Gateway_1" targetRef="End">
      <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">Task2.action == confirm</bpmn:conditionExpression>
    </bpmn:sequenceFlow>
    <bpmn:sequenceFlow id="SequenceFlow_5" sourceRef="Gateway_1" targetRef="Task1">
      <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">Task2.action == reject</bpmn:conditionExpression>
    </bpmn:sequenceFlow>
  </bpmn:process>
</bpmn:definitions>
```

TL;DR; extracted the flows exiting the gateway from the bpmn abov:
```xml
<bpmn:sequenceFlow id="SequenceFlow_4" sourceRef="Gateway_1" targetRef="End">
    <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">Task2.action == confirm</bpmn:conditionExpression>
</bpmn:sequenceFlow>
<bpmn:sequenceFlow id="SequenceFlow_5" sourceRef="Gateway_1" targetRef="Task1">
    <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">Task2.action == reject</bpmn:conditionExpression>
</bpmn:sequenceFlow>
```

**Comment by RonnyB71** [comment]
See requirements from: [#271](https://github.com/Altinn/app-frontend-react/issues/271) and #137
